src/icon/checkIcon.tsx

```diff
@@ -24,23 +27,29 @@ const CheckIcon = (props: CheckProps) => (
       },
     ])}
   >
-    <Fragment>
-      <path
-        d="M6.5 12.5l4 4 8-8"
-        fill="none"
-        stroke={props.iconColor}
-        strokeWidth={props.thin ? '1' : '2'}
-        strokeLinecap="round"
-        strokeLinejoin="round"
-        strokeMiterlimit="10"
-      />
-    </Fragment>
+    <path
+      d="M6.5 12.5l4 4 8-8"
+      fill="none"
+      stroke={props.iconColor}
+      strokeWidth={props.thin ? '1' : '2'}
+      strokeLinecap="round"
+      strokeLinejoin="round"
+      strokeMiterlimit="10"
+    />
   </BaseIcon>
 )
 
 export const StyledCheckIcon = styled(CheckIcon)`
+  /* .kirk-icon--disabled forces a stroke color, but it needs to be overriden if a specific background is passed */
+  &.kirk-icon--disabled path {
+    stroke: ${props =>
+      props.backgroundColor !== defaultBackgroundColor ? color.white : color.fadedText};
+  }
   & {
-    background-color: ${props => props.backgroundColor};
+    background-color: ${props =>
+      props.isDisabled && props.backgroundColor !== defaultBackgroundColor
```

Comment:
cool ! you are already using this model here! why not use everywhere else?

Reply:
The checkIcon is already an exception, the only one with a "backgroundColor" prop. It was a mess to change correctly so I'm not sure this is ideal.

I'm even thinking we could split this Icon into "just a check icon" and "circled check icon".

Reply:
Typography and TheVoice uses also **styled-components**. Since styled-components is our official stack, we should start writting with its syntax. 

It will keep your code DRY and easy to maintain

Reply:
Done